infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/ExternalServerPathDemangler.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.external;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.inject.Named;
+import org.eclipse.che.commons.annotation.Nullable;
+
+public class ExternalServerPathDemangler {
+  private final Pattern pathTransformInverse;
+
+  @Inject
+  public ExternalServerPathDemangler(
+      @Nullable @Named("che.infra.kubernetes.ingress.path_transform") String pathTransformFmt) {
+    this.pathTransformInverse = extractPathFromFmt(pathTransformFmt);
+  }
+
+  private static Pattern extractPathFromFmt(String pathTransformFmt) {
+    int refIdx = pathTransformFmt.indexOf("%s");
+    String matchPath = "(.*)";
```

Comment:
This could have a more descriptive name, and perhaps be a named group -- as it stands it's easy to confuse this with the default value for `che.infra.kubernetes.ingress.path_transform`